examples/material-ui/app/entry.server.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { renderToString } from 'react-dom/server';
+import { RemixServer } from 'remix';
+import type { EntryContext } from 'remix';
+
+import createEmotionCache from './src/createEmotionCache';
+import theme from './src/theme';
+
+import CssBaseline from '@mui/material/CssBaseline';
+import { ThemeProvider } from '@mui/material/styles';
+import { CacheProvider } from '@emotion/react';
+import createEmotionServer from '@emotion/server/create-instance';
+
+export default function handleRequest(
+  request: Request,
+  responseStatusCode: number,
+  responseHeaders: Headers,
+  remixContext: EntryContext,
+) {
+  const cache = createEmotionCache();
+  const { extractCriticalToChunks } = createEmotionServer(cache);
+
+  const MuiRemixServer = () => (
+    <CacheProvider value={cache}>
+      <ThemeProvider theme={theme}>
+        {/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
+        <CssBaseline />
+        <RemixServer context={remixContext} url={request.url} />
+      </ThemeProvider>
+    </CacheProvider>
+  );
+
+  // Render the component to a string.
+  const html = renderToString(<MuiRemixServer />);
+
+  // Grab the CSS from emotion
+  const { styles } = extractCriticalToChunks(html);
+
+  let stylesHTML = '';
+
+  styles.forEach(({ key, ids, css }) => {
+    const emotionKey = `${key} ${ids.join(' ')}`;
+    const newStyleTag = `<style data-emotion="${emotionKey}">${css}</style>`;
+    stylesHTML = `${stylesHTML}${newStyleTag}`;
+  });
+
+  // Add the emotion style tags after the insertion point meta tag
+  const markup = html.replace(
+    /<meta(\s)*name="emotion-insertion-point"(\s)*content="emotion-insertion-point"(\s)*\/>/,
```

Comment:
I think this could be much simpler if this were something like `__STYLES__` (like the styled-components example).

Reply:
@kentcdodds I think it is correct because in the styled-components example the __STYLES__ was replaced from the extracted CSS and in material-UI, we need to add the emotion style tags after the insertion point meta tags and this is a one-time thing as shown in the official material-UI examples.